pkg/scheduler/predicates.go

```diff
@@ -255,6 +255,22 @@ func NewServiceAffinityPredicate(podLister PodLister, serviceLister ServiceListe
 	return affinity.CheckServiceAffinity
 }
 
+// TODO:@sdminonne: remove it once checked with @agupta
+func copyValueIfKeyEquals(key string, affinityLabels map[string]string, nodeSelector labels.Selector) bool {
```

Comment:
@abhgupta OK. I'm going to handle all the operators and send you back the code for review.
@brendandburns I'll introduce the `findValue`.